src/hotspot/share/runtime/arguments.hpp

```diff
@@ -462,9 +462,8 @@ class Arguments : AllStatic {
   static int  PropertyList_readable_count(SystemProperty* pl);
 
   static bool is_internal_module_property(const char* option);
-  static bool is_add_modules_property(const char* key);
-  static unsigned int addmods_count() { return  _addmods_count; }
-  static bool is_module_path_property(const char* key);
+  static bool is_non_cds_compatible_internal_module_property(const char* property);
+  static bool internal_module_property_helper(const char* property, bool check_for_cds);
```

Comment:
Instead of `bool check_for_cds`, how about `bool check_all`?
If you choose to rename that, you'll need to change the checking of the bool inside the function and also flip the bool value at the callsites.

Reply:
I don't think `check_all` is a good name -- it's not clear what the opposite of `check_all` means.

`internal_module_property_helper()` should be a private function whose definition. It should be used only by arguments.cpp and not be a public function.